src/leaking.md

```diff
@@ -53,49 +120,83 @@ library:
 
 ## Drain
 
+<!--
 `drain` is a collections API that moves data out of the container without
 consuming the container. This enables us to reuse the allocation of a `Vec`
 after claiming ownership over all of its contents. It produces an iterator
 (Drain) that returns the contents of the Vec by-value.
+-->
 
+`drain` は、コンテナを消費せずにコンテナからデータをムーブする、
+コレクションの API です。これによって、 `Vec` の全ての内容の所有権を獲得した後に、 `Vec` の
+アロケーションを再利用することが出来ます。 `drain` は Vec の内容を値で返すイテレータ (Drain) を
+生成します。
+
+<!--
 Now, consider Drain in the middle of iteration: some values have been moved out,
 and others haven't. This means that part of the Vec is now full of logically
 uninitialized data! We could backshift all the elements in the Vec every time we
 remove a value, but this would have pretty catastrophic performance
 consequences.
+-->
+
+では、 Drain がイテレーションの真っ最中であるとしましょう。ムーブされた値もあれば、
+まだの値もあります。つまりこれは、 Vec の一部のデータが今、論理的には未初期化のデータで
+埋まっていることを意味します! 値を削除する度に Vec の要素を後ろにずらすことも出来たでしょう。
```

Comment:
it may be only me, but 「後ろにずらす」sounds like moving elements towards the end of `Vec` whereas what `backshift` means is to move elements towards lower index. 
 Maybe just 「ずらす」?

Reply:
:+1: